File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/AbstractDomainTest.java
package com.coupang.springframework.data.requery.domain;

import com.coupang.springframework.data.requery.core.RequeryOperations;
import com.coupang.springframework.data.requery.configs.RequeryTestConfiguration;
import io.requery.sql.EntityDataStore;
import lombok.extern.slf4j.Slf4j;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import javax.inject.Inject;
import java.util.Random;

import static org.assertj.core.api.Assertions.assertThat;


/**
 * com.coupang.springframework.data.requery.domain.AbstractDomainTest
 *
 * @author debop
 * @since 18. 6. 4
 */
@Slf4j
@RunWith(SpringRunner.class)
@SpringBootTest(classes = { RequeryTestConfiguration.class })
public abstract class AbstractDomainTest {

    protected static Random rnd = new Random(System.currentTimeMillis());

    @Inject
    protected EntityDataStore<Object> dataStore;

    @Inject
    protected RequeryOperations requeryTemplate;

    @Test
    public void context_loading() {
        assertThat(dataStore).isNotNull();
        assertThat(requeryTemplate).isNotNull();
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/AbstractDomainTest.kt
package com.coupang.springframework.data.requery.domain

import com.coupang.kotlinx.logging.KLogging
import com.coupang.springframework.data.requery.AbstractSpringDataRequeryTest
import com.coupang.springframework.data.requery.configs.RequeryTestConfiguration
import com.coupang.springframework.data.requery.core.RequeryKotlinTemplate
import com.coupang.springframework.data.requery.core.RequeryTemplate
import io.requery.Persistable
import io.requery.sql.EntityDataStore
import io.requery.sql.KotlinEntityDataStore
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner
import java.util.*
import javax.inject.Inject

/**
 * AbstractDomainTest
 *
 * @author debop@coupang.com
 * @since 18. 5. 31
 */
@RunWith(SpringRunner::class)
@SpringBootTest(classes = [RequeryTestConfiguration::class])
abstract class AbstractDomainTest: AbstractSpringDataRequeryTest() {

    companion object: KLogging() {
        val rnd = Random(System.currentTimeMillis())
    }


    @Inject lateinit var dataStore: EntityDataStore<Persistable>
    @Inject lateinit var ktDataStore: KotlinEntityDataStore<Persistable>
    @Inject lateinit var requeryTmpl: RequeryTemplate
    @Inject lateinit var requeryKtTmpl: RequeryKotlinTemplate

    @Test
    fun `context loading`() {
        assertThat(dataStore).isNotNull
        assertThat(ktDataStore).isNotNull
        assertThat(requeryTmpl).isNotNull
        assertThat(requeryKtTmpl).isNotNull
    }

}